src/web-rtc-client.js

```diff
@@ -833,70 +834,106 @@ export default class WebRTCClient extends Emitter {
     }
   }
 
-  changeVideoInputDevice(id: string, session: ?Inviter) {
+  changeVideoInputDevice(id: ?string, session: ?Inviter) {
+    this.setVideoInputDevice(id);
+
+    if (session) {
+      return this.changeSessionVideoInputDevice(id, session);
+    }
+  }
+
+  setVideoInputDevice(id: ?string) {
     const currentId = this.getVideoDeviceId();
     if (id === currentId) {
       return null;
     }
 
     // let's update the local value
-    if (this.video && this.video.deviceId) {
-      // $FlowFixMe
-      this.video.deviceId.exact = id;
+    if (this.video) {
+      this.video = {
```

Comment:
This may override other video constraints, no?

Reply:
Yes that's the point. Here we want only to change the video device id (without changing it in the session), so we have to change the constraint.
Changing the constraint in the session is done in `changeVideoInputDevice` .